homeassistant/components/modbus/binary_sensor.py

```diff
@@ -84,14 +84,13 @@ async def async_setup_platform(
             CONF_NAME: "no name",
             CONF_BINARY_SENSORS: config[CONF_INPUTS],
         }
-        config = None
 
     for entry in discovery_info[CONF_BINARY_SENSORS]:
-        if CONF_HUB in entry:
-            # from old config!
-            hub: ModbusHub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
-        else:
-            hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        hub: ModbusHub = (
+            hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
+            if CONF_HUB in entry
+            else hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        )
```

Comment:
Declaring `hub_name` without a type doesn't matter now since we're not passing it to a typed context.
```suggestion
        hub_name = entry[CONF_HUB] if CONF_HUB in entry else discovery_info[CONF_NAME]
        hub: ModbusHub = hass.data[MODBUS_DOMAIN][hub_name]
```

Reply:
@janiversen why did you do the opposite from what was suggested to you by several people?

Reply:
It is very seldom that I dont agree with your suggestions.....this is one of the few.

By simply removing the type from hub I am back to my original PR, that fits with your argument and is not wrong ... and most importantly it is accepted by mypy.

Reply:
@KapJI because I do not see most of the suggestions as logical, I prefer a much similar solution.

Of course the nuclear option is to close the PR and keep having modbus excluded....I have choosen not to do that.